src/picongpu/include/particles/traits/GetEffectiveAtomicNumbers.hpp

```diff
@@ -42,6 +43,15 @@ struct GetEffectiveAtomicNumbers
     typedef typename GetFlagType<FrameType,effectiveAtomicNumbers<> >::type FoundEffectiveAtomicNumbersAlias;
     /* Extract vector of effective atomic numbers */
     typedef typename PMacc::traits::Resolve<FoundEffectiveAtomicNumbersAlias >::type type;
+
+    static constexpr int protonNumber = static_cast<int>(GetAtomicNumbers<SpeciesType>::type::numberOfProtons);
+    /* length of the ionization energy vector */
+    static constexpr int vecLength = type::dim;
+    /* assert that the number of arguments in the vector equal the proton number */
+    PMACC_CASSERT_MSG(
+        __The_given_number_of_effective_core_charge_numbers_should_be_exactly_the_proton_number_of_the_species__,
```

Comment:
I would skip "core" here and only call it "effective atomic number Z_eff" because that is the precise terminus